RecoLocalTracker/SiPhase2Clusterizer/src/Phase2TrackerClusterizerAlgorithm.cc

```diff
@@ -0,0 +1,96 @@
+#include "RecoLocalTracker/SiPhase2Clusterizer/interface/Phase2TrackerClusterizerAlgorithm.h"
+
+#include "Geometry/CommonTopologies/interface/PixelTopology.h"
+
+/*
+ * Initialise the clusterizer algorithm
+ */
+
+Phase2TrackerClusterizerAlgorithm::Phase2TrackerClusterizerAlgorithm(unsigned int maxClusterSize, unsigned int maxNumberClusters) : maxClusterSize_(maxClusterSize), maxNumberClusters_(maxNumberClusters), nrows_(0), ncols_(0) { }
+
+/* 
+ * Change the size of the 2D matrix for this module (varies from pixel to strip modules)
+ */
+
+void Phase2TrackerClusterizerAlgorithm::setup(const PixelGeomDetUnit* pixDet) {
+    const PixelTopology& topol(pixDet->specificTopology());
+    nrows_ = topol.nrows();
+    ncols_ = topol.ncolumns();
+    matrix_.setSize(nrows_, ncols_);
+}
+
+/* 
+ * Go over the Digis and create clusters
+ */
+
+void Phase2TrackerClusterizerAlgorithm::clusterizeDetUnit(const edm::DetSet< Phase2TrackerDigi >& digis, edmNew::DetSetVector< Phase2TrackerCluster1D >::FastFiller& clusters) {
+
+    // Fill the 2D matrix with the ADC values
```

Comment:
Please change the comment to avoid misunderstandings... there is no more ADC values.